### PR TITLE
Compartmentalize moonworm dependencies

### PR DESCRIPTION
**Pull request by zomglings**
## Changes

Updated dependencies to not be so heavy.

- Created `moonworm[moonstream]` for the Moonstream DB-related dependencies
- Moved `web3[tester]` into `dev` dependencies.
- Top-level dependencies only include `web3` (with no lower bound)
- Removed `pysha3` dependency at top level (not sure why that was there in the first place)

## How to test these changes?

The tests were already failing before these changes. I tested code generation but not crawling.

We should fix the tests or remove the ones that don't matter. (Separate issue/PR)

### CODE REVIEW

**Comment by zomglings** [comment]
I checked that the `mooncrawl` crawler on https://github.com/bugout-dev/moonstream pins the `moonworm` version to `0.1.11`. I am concerned that we may break dependencies elsewhere @Andrei-Dolgolev , @kompotkot .

In the future, in Moonstream crawlers, we should use `moonworm[moonstream]` as dependency, not just `moonworm`.

File: setup.py [hunk 1]
```diff
@@ -15,18 +15,12 @@
         "black",
         "inflection",
         "libcst",
-        "moonstreamdb",
-        "pysha3<2.0.0,>=1.0.0",
         "tqdm",
-        "typing-extensions<4,>=3.7.4",
-        "web3[tester] >=5.27.0",
+        "web3",
```
Comment (Yhtiyar) [inline setup.py:19]: web3 >=5.27.0

File: setup.py [hunk 2]
```diff
@@ -15,18 +15,12 @@
         "black",
         "inflection",
         "libcst",
-        "moonstreamdb",
-        "pysha3<2.0.0,>=1.0.0",
```
Comment (Yhtiyar) [inline setup.py:19]: pysha3 was necessary for something. (Not sure if it was on crawler side)

File: setup.py [hunk 3]
```diff
@@ -15,18 +15,12 @@
...
         "tqdm",
-        "typing-extensions<4,>=3.7.4",
```
Comment (Yhtiyar) [inline setup.py:21]: this one was also added for some reason. dunno remember.

**Review by Yhtiyar — CHANGES_REQUESTED** [review]
I am gonna merge this only after I will add tests for crawling logic.

**Comment by Yhtiyar** [comment]
closing in favor of #83

**Pull request closed without merging**